### PR TITLE
Create github_delete action_invoked.yaml

### PR DESCRIPTION
**Pull request by faisalusuf**
Hello,

I am creating a series of Detection rules (that's being used in our prod environment) for GitHub cloud and server audit logs. Keeping in view of recent breaches where code repositories were compromised, when we started looking for relevant Sigma rules there were none. 
Therefore we created it from the scratch and tested it over months and mature it in a prod environment. It is time to contribute back to the community. This is the first rule many other rules will be pushed soon.

This is my first ever contribution to Sigma repo.

Thanks
Faisal.

### CODE REVIEW

**Comment by frack113** [comment]
As you add a new logsource, you need to update [logsource.json](https://github.com/SigmaHQ/sigma/blob/master/tests/logsource.json) too for the test
```
        "gcp":{
            "commun": [],
            "empty": [],
            "category":{},
            "service":{
                "gcp.audit":[]
            }
        },
        "github":{
            "commun": [],
            "empty": [],
            "category":{},
            "service":{
                "audit_logs":[]
            }
        },
        "google_workspace":{
            "commun": [],
            "empty": [],
            "category":{},
            "service":{
                "google_workspace.admin":[]
            }
        },
```
Thanks

File: rules/cloud/github/github_delete action_invoked.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    - https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/reviewing-the-audit-log-for-your-organization#audit-log-actions
+tags:
+    - attack.impact
+    - attack.t1213.003
+logsource:
+    product: github
+    service: audit_logs
+detection:
+    selection:
+        action:
```
Comment (frack113) [inline rules/cloud/github/github_delete action_invoked.yaml:17]:
```suggestion
        operation_type: 'remove'
        action:
```

File: rules/cloud/github/github_new_secret_created.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    - https://docs.github.com/en/organizations/keeping-your-organization-secure/managing-security-settings-for-your-organization/reviewing-the-audit-log-for-your-organization#audit-log-actions
+tags:
+    - attack.t1078
+    - attack.t1078.004
+logsource:
+    product: github
+    service: audit_logs
+detection:
+    selection:
+        action:
```
Comment (frack113) [inline rules/cloud/github/github_new_secret_created.yml:17]:
```suggestion
        operation: 'create'
        action:
```

File: rules/cloud/github/github_outside_collaborator_detected.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+tags:
+    - attack.t1098.001
+    - attack.t1213.003
+    - attack.t1098.003
+logsource:
+    product: github
+    service: audit_logs
+detection:
+    selection:
+        action:
```
Comment (frack113) [inline rules/cloud/github/github_outside_collaborator_detected.yml:19]:
```suggestion
        operation: 'authentication'
        action:
```

**Review by frack113 — COMMENTED** [review]
I add the operation form ref
```
Use the operation qualifier to limit actions to specific types of operations. For example:

    operation:access finds all events where a resource was accessed.
    operation:authentication finds all events where an authentication event was performed.
    operation:create finds all events where a resource was created.
    operation:modify finds all events where an existing resource was modified.
    operation:remove finds all events where an existing resource was removed.
    operation:restore finds all events where an existing resource was restored.
    operation:transfer finds all events where an existing resource was transferred.
```

**Comment by nasbench** [comment]
> "github":{
>             "commun": [],
>             "empty": [],
>             "category":{},
>             "service":{
>                 "audit_logs":[]
>             }
>         },

I did that so no need to now.

**Comment by faisalusuf** [comment]
@frack113 we are good here, thanks for adding additional context.

**Pull request merged**